web/src/recent_view_ui.ts

```diff
@@ -660,7 +776,7 @@ export function filters_should_hide_topic(topic_data) {
         return true;
     }
 
-    if (dropdown_filters.has(views_util.FILTERS.UNMUTED_TOPICS) && topic_data.type === "stream") {
+    if (dropdown_filters.has(views_util.FILTERS.UNMUTED_TOPICS) && msg.type === "stream") {
```

Comment:
So I think this is the wrong conversion. We are currently mostly looking at `msg.stream_id` and `msg.topic` rather than the similar elements on the `topic_data` object today, so it's a smaller diff to move over this assert, but I think we should be passing the `topic_data.stream_id / topic_data.topic` into this function.

The motivation is that when we're able to provide a dedicated efficient API for the recent view, it's probably not going to preserve the invariant that any topic in this view has a message in `message_store`.

Reply:
I didn't get to this today, but I'll push the rest of the changes. I think it would be fine to do this as a followup change.

Reply:
Yeah, you can do that as a near-term follow-up.